dali/operators/reader/numpy_reader_op.h

```diff
@@ -58,36 +62,39 @@ class NumpyReader : public DataReader<Backend, Target> {
 
     int batch_size = GetCurrBatchSize();
     const auto& file_0 = GetSample(0);
-    TypeInfo output_type = file_0.data.type();
-    int ndim = file_0.data.shape().sample_dim();
+    TypeInfo output_type = file_0.get_type();
+    int ndim = file_0.get_shape().sample_dim();
     TensorListShape<> sh(batch_size, ndim);
 
-    bool need_slice = slice_attr_.template ProcessArguments<Backend>(ws, batch_size, ndim);
-
+    bool has_roi_args = slice_attr_.template ProcessArguments<Backend>(ws, batch_size, ndim);
     rois_.clear();
-    if (need_slice)
+    if (has_roi_args)
       rois_.resize(batch_size);
 
+    need_copy_.clear();
+    need_transpose_.clear();
+    need_slice_.clear();
+    need_slice_perm_.clear();
```

Comment:
That's a terrible idea. `std::set` is a tree - it will a new node upon each insert and deallocate multiple nodes upon each clear.